File: Java/CodingBat/closeFar.java
// http://codingbat.com/prob/p138990

public boolean closeFar(int a, int b, int c) {
  int abAbs = Math.abs(a-b);
  int acAbs = Math.abs(a-c);
  int bcAbs = Math.abs(b-c);
  
  // totally readable code right here
  return ((abAbs <= 1 ^ acAbs <= 1) && ((abAbs >= 2 && bcAbs >= 2) ^ (acAbs >= 2 && bcAbs >= 2)));
}

File: Python/CodingBat/close_far.py
# http://codingbat.com/prob/p160533

def close_far(a, b, c):
  ab_diff = abs(a-b)
  ac_diff = abs(a-c)
  bc_diff = abs(b-c)
  
  if ab_diff <= 1:
    if bc_diff >= 2 and ac_diff >= 2:
      return True
    
  if ac_diff <= 1:
    if bc_diff >= 2 and ab_diff >= 2:
      return True
  
  return False
